src/vs/workbench/contrib/chat/browser/chatPasteProviders.ts

```diff
@@ -136,13 +152,126 @@ export function isImage(array: Uint8Array): boolean {
 	);
 }
 
+export class CopyTextProvider implements DocumentPasteEditProvider {
+
+	static readonly id = 'text';
+	static readonly kind = new HierarchicalKind('text.plain');
+
+	readonly id = PasteTextProvider.id;
+	readonly kind = PasteTextProvider.kind;
+
+	readonly copyMimeTypes = [Mimes.text, 'vscode-editor-data'];
+
+	async prepareDocumentPaste(model: ITextModel, ranges: readonly IRange[], dataTransfer: IReadonlyVSDataTransfer, token: CancellationToken): Promise<undefined | IReadonlyVSDataTransfer> {
+		if (model.uri.scheme === ChatInputPart.INPUT_SCHEME) {
+			return;
+		}
+		const customDataTransfer = new VSDataTransfer();
+		const rangesString = JSON.stringify({ ranges: ranges[0], uri: model.uri.toString() });
+		customDataTransfer.append('additional-editor-data', createStringDataTransferItem(rangesString));
+		return customDataTransfer;
+	}
+}
+
+export class PasteTextProvider implements DocumentPasteEditProvider {
+
+	static readonly id = 'text';
+	static readonly kind = new HierarchicalKind('text.plain');
```

Comment:
Make sure to update these ids/kinds. Something like `chat.attach.text`. You can actually remove `id` entirely. It's only used for a very specific case with the default text paste provider